src/RestApi.ts

```diff
@@ -101,16 +106,22 @@ export class RestApi {
      * @returns {Promise<any>} - Returns a Promise that resolves with the answer from the rest-api in a js object, or rejects if an error occurs.
      */
     public doRequest(url: string, method: string, data: any): Promise<any> {
+        // determine version to use, if not yet determined, and then request
+        const versionPromise = this._contentType ? Promise.resolve() : this.determineApiVersion();
```

Comment:
I would move this logic to a getApiVersion function that returns a promise, which also does the determineApiVersion logic if necessary.

That 'isolates' the logic required to get an api version, regardless of whether that has happened already or not.

Reply:
Yes, done